### ISSUE TITLE
Explorer tree view does not maintain focus after adding file or directory

### ISSUE THREAD

**Issue by dsifford** [Microsoft/vscode#41559]
```sh
$ code --version
1.19.2
490ef761b76b3f3b3832eff7a588aac891e5fe80
x64
$ uname -a
Linux archlinux 4.14.13-1-ARCH #1 SMP PREEMPT Wed Jan 10 11:14:50 UTC 2018 x86_64 GNU/Linux
```

**Assumptions:**

```json
    {
        "key": "a",
        "command": "explorer.newFile",
        "when": "explorerViewletVisible && filesExplorerFocus && !inputFocus"
    },
```

**Gist:** When navigating the file tree with the keyboard, after adding a new file using the shortcut above, after the enter key is pressed (thus, creating the file with the name specified), the focus is not returned back to the file tree. 

Instead, to resume navigating the tree with the keyboard, you have to first click the tree somewhere again.

**Steps to Reproduce:**

1. Create a new file using a shortcut similar to the one above.
2. Notice that focus is not returned when the enter key is pressed

**Does this issue occur when all extensions are disabled?:** Yes

**Comment by isidorn** [comment]
Thanks for the nice repro steps.
When a new folder gets created the focus simply disapears and that is definetly a bug. However it is debatable if when a new file is created the focus should be passed to the file or stay in the explorer.

I could fix this by removing the check [here](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/parts/files/electron-browser/views/explorerViewer.ts#L291)
And by passing `preserveFocus` [here](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/parts/files/electron-browser/fileActions.ts#L574)

However the first check was there for a reason, some earlier bugfix thus assigning also @bpasero to get his opinion

**Comment by bpasero** [comment]
I think we could "fix" this but I would say there will be a lot of people that will then complain why they have to click into the editor to start typing after creating a new file. I would not change this behaviour and rather have a separate command to create a new file and keep focus.

**Comment by dsifford** [comment]
Thanks for looking into this @isidorn 

@bpasero Good point. Come to think of it, I think losing focus after creating directories was the only circumstance where I was consistently annoyed (I've got that bound to `shift+a`).

**Comment by isidorn** [comment]
The directories focus lost I can look into fixing -> january

**Comment by dsifford** [comment]
Hey @isidorn while we're on the topic of focus issues, I noticed another one tonight. (Let me know if you'd prefer this taken to a separate issue).

This one is even more annoying IMHO. Here's the gist:

1. Open any file that has support for "Go to next error or warning" (e.g., TypeScript).
2. Make type out some code that has a few warnings in it.
3. Press the keyboard shortcut to take you to the next warning (default is `F8`).
4. Press the `Esc` key.

**What happens:** Nothing. Error message still visible.

**What I'd expect to happen:** Escape should close the warning message. 

In order for `Esc` to toggle the visibility state of the error or warning, I have to manually click inside the warning message before pressing the escape key.

**Comment by isidorn** [comment]
@dsifford this is by design for accessibility reasons. Namely users using screen readers use F8 to go through all their errors and they expect the focus to stay in the editor.
If you want to discuss this further feel free to create a new issue

**Comment by isidorn** [comment]
@dsifford I have pushed a fix for the folder creation. For files we would consider this if more users complain, for now keeping as it is.
@bpasero now I am setting focus based if it is a blur event or not, seems to work nicer than the previous fix and this fixes this issue and the #20269
